pkg/util/kubernetes/apiserver/controllers.go

```diff
@@ -56,15 +62,28 @@ type ControllerContext struct {
 
 // StartControllers runs the enabled Kubernetes controllers for the Datadog Cluster Agent. This is
 // only called once, when we have confirmed we could correctly connect to the API server.
-func StartControllers(ctx ControllerContext) error {
+func StartControllers(ctx ControllerContext) {
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(controllerCatalog))
+	defer close(errChan)
 	for name, cntrlFuncs := range controllerCatalog {
 		if !cntrlFuncs.enabled() {
 			log.Infof("%q is disabled", name)
 			continue
 		}
-		err := cntrlFuncs.start(ctx)
+
+		// controllers should be started in parallel as their start functions are
+		// blocking until the informers are sync'ed or the sync period timed-out.
+		// for error propagation we rely on a buffered channel to gather errors
+		// from the spawned goroutines.
+		wg.Add(1)
+		go cntrlFuncs.start(ctx, &wg, errChan)
+	}
+
+	wg.Wait()
+	for err := range errChan {
 		if err != nil {
-			log.Errorf("Error starting %q: %s", name, err.Error())
+			log.Warnf("Error while starting controller: %v", err)
 		}
 	}
```

Comment:
Maybe we can use a `errs := []error` and use to after https://godoc.org/k8s.io/apimachinery/pkg/util/errors#NewAggregate

```golang
return errors.NewAggregate(errs)
```